flink-end-to-end-tests/test-scripts/common_kubernetes.sh

```diff
@@ -87,6 +94,35 @@ function setup_kubernetes_for_linux {
     sudo sysctl fs.protected_regular=0
 }
 
+function retry_download {
+    if [[ "$#" != 3 ]]; then
+       echo "Fatal error: No parameter or too many parameters passed: $@"
+       exit 1;
+    fi
+
+    local download_type download_options download_url download_command
+    download_type="$1"
```

Comment:
Why do we have to support `curl` and `wget`. They are both doing the same thing (i.e. downloading artifacts). Or what's the reason why you're sticking to both? AFAIU, we could just pick one of the two and implement the artifact download in this `retry_download` function without additional parameters `download_type` and `download_options`. Or am I missing something? :thinking: